File: sonarqube-companion-rest/src/main/java/pl/consdata/ico/sqcompanion/config/ServerDefinition.java
package pl.consdata.ico.sqcompanion.config;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class ServerDefinition {

    private String id;
    private String url;
    private ServerAuthentication authentication;

    public boolean hasAuthentication() {
        return authentication != null
                && authentication.getType() != null
                && !authentication.getType().isEmpty();
    }

}

File: sonarqube-companion-frontend/src/app/settings/model/server-definition.ts
import {AuthenticationData} from './authentication-data';

export class ServerDefinition {
  id: string;
  url: string;
  authentication: AuthenticationData;
  blacklistUsers: string[];


  constructor(data?: any) {
    if (data) {
      this.id = data.id;
      this.url = data.url;
      this.authentication = new AuthenticationData(data.authentication);
      this.blacklistUsers = data.blacklistUsers;
    } else {
      this.authentication = new AuthenticationData();
    }
  }


}
